### PR TITLE
Replace self defined const StatusTooManyRequests with http.StatusTooM…

### PR DESCRIPTION
**Pull request by jfbai**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Replace the self defined const `StatusTooManyRequests` with `http.StatusTooManyRequests` to make status code more consistent.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
Please take a look at the error message:

```go
# k8s.io/kubernetes/plugin/pkg/admission/eventratelimit [k8s.io/kubernetes/plugin/pkg/admission/eventratelimit.test]
plugin/pkg/admission/eventratelimit/admission_test.go:519:43: undefined: "k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/api/errors".StatusTooManyRequests
```

**Comment by jfbai** [comment]
> Please take a look at the error message:
> 
> ```go
> # k8s.io/kubernetes/plugin/pkg/admission/eventratelimit [k8s.io/kubernetes/plugin/pkg/admission/eventratelimit.test]
> plugin/pkg/admission/eventratelimit/admission_test.go:519:43: undefined: "k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/api/errors".StatusTooManyRequests
> ```

@draveness Fixed.

**Comment by jfbai** [comment]
@dims PTAL, thanks.

**Review by draveness — APPROVED** [review]
/lgtm

Thanks @jfbai

**Comment by jfbai** [comment]
> /lgtm
> 
> Thanks @jfbai

Thanks @draveness :)

File: staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go [hunk 1]
```diff
@@ -29,12 +29,6 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
-const (
-	// StatusTooManyRequests means the server experienced too many requests within a
-	// given window and that the client must wait to perform the action again.
-	StatusTooManyRequests = 429
```
Comment (yue9944882) [inline staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go]: stripping this from staging area straightly breaks backward-compatibility

/hold

File: staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go [hunk 2]
```diff
@@ -32,6 +32,8 @@ import (
 const (
 	// StatusTooManyRequests means the server experienced too many requests within a
 	// given window and that the client must wait to perform the action again.
+	// DEPRECATED: please use http.StatusTooManyRequests, this will be removed in
+	// the future version.
 	StatusTooManyRequests = 429
```
Comment (yliaog) [inline staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go]: what about replacing 429 with http.StatusTooManyRequests?

File: staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go (hunk 1 above)
Comment (jfbai) [reply staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go]: make sense, fixed.

File: staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go (hunk 2 above)
Comment (jfbai) [reply staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go]: make sense, fixed.

**Comment by jfbai** [comment]
@yue9944882 @yliaog PTAL, thanks a lot.

**Comment by jfbai** [comment]
ping @yue9944882 to review and remove `do-not-merge/hold` label.

**Comment by yue9944882** [comment]
sry, notification is flooding

/hold cancel

**Comment by jfbai** [comment]
ping @liggitt to help review and approve, thanks a lot. :)

**Pull request merged**